3.8/aql/fundamentals-limits.md

```diff
@@ -0,0 +1,21 @@
+---
+layout: default
+description: Known limitations that apply in AQL queries
+---
+Known limitations for AQL queries
+=================================
+
+The following limitations are known for AQL queries:
+
+- An AQL query cannot use more than _1000_ result registers. One
+  result register is needed for every named query variable and for 
+  internal/anonymous query variables.
```

Comment:
As since 3.7 we can reuse a register after a variable is no longer needed, this sounds more pessimistic than is actually the case for most queries. I'm not sure though whether we want to go into more detail or not.

Reply:
Although you are right that some registers may be reused, I wouldn't document this as a "guaranteed" feature.
Queries using close to 1000 registers will suffer from various issues anyway.